src/App.tsx

```diff
@@ -1,26 +1,175 @@
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorTodos(ErrorType.Loading);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const onAddTodo = async (newTodo: Omit<Todo, 'id'>): Promise<void> => {
+    try {
+      const createdTodo = await createTodo(newTodo);
+
+      setTodos(currentTodos => [...currentTodos, createdTodo]);
+    } catch (error) {
+      setErrorTodos(ErrorType.Add);
+      throw error;
+    }
+  };
+
+  const onDeleteTodo = async (todoId: number) => {
+    setLoadingTodosIds(currentIds => [...currentIds, todoId]);
+    try {
+      await deleteTodo(todoId);
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+    } catch (error) {
+      setErrorTodos(ErrorType.Delete);
+      throw error;
+    } finally {
+      setLoadingTodosIds(currentIds => currentIds.filter(id => id !== todoId));
+    }
+  };
+
+  const onDeleteAllCompleted = () => {
+    completedTodos.forEach(completedTodo => {
+      onDeleteTodo(completedTodo.id);
+    });
+  };
+
+  const onUpdateTodo = async (newTodo: Todo) => {
+    setLoadingTodosIds(currentIds => [...currentIds, newTodo.id]);
+    try {
+      const updatedTodo = await updateTodo(newTodo);
+
+      setTodos(currentTodos =>
+        currentTodos.map(todo => {
+          return todo.id === updatedTodo.id ? updatedTodo : todo;
+        }),
+      );
+    } catch (error) {
+      setErrorTodos(ErrorType.Update);
+      throw error;
+    } finally {
+      setLoadingTodosIds(currentIds =>
+        currentIds.filter(id => id !== newTodo.id),
+      );
+    }
+  };
+
+  const onUpdateToggleAll = () => {
+    todos.forEach(todo => {
+      if (todos.length === completedTodos.length) {
+        onUpdateTodo({ ...todo, completed: false });
+
+        return;
+      }
+
+      if (todo.completed === false) {
+        onUpdateTodo({ ...todo, completed: true });
+
+        return;
+      }
+    });
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          todos={todos}
+          completedTodosCount={completedTodosCount}
+          onAddTodo={onAddTodo}
+          setErrorTodos={setErrorTodos}
+          tempTodo={tempTodo}
+          setTempTodo={setTempTodo}
+          onUpdateToggleAll={onUpdateToggleAll}
+        />
+
+        {todos.length > 0 && (
```

Comment:
better option(it will be converted to boolean value)
```suggestion
        {!!todos.length && (
```